promql/promqltest/testdata/native_histograms.test

```diff
@@ -784,3 +784,42 @@ eval_warn instant at 1m rate(some_metric[30s])
 # Start with exponential, end with custom.
 eval_warn instant at 30s rate(some_metric[30s])
     # Should produce no results.
+
+# Histogram with constant buckets.
+load 1m
+    const_histogram {{schema:0 sum:1 count:1 buckets:[1 1 1]}} {{schema:0 sum:1 count:1 buckets:[1 1 1]}} {{schema:0 sum:1 count:1 buckets:[1 1 1]}} {{schema:0 sum:1 count:1 buckets:[1 1 1]}} {{schema:0 sum:1 count:1 buckets:[1 1 1]}}
+
+# There is no change to the bucket count over time, thus rate is 0 in each bucket.
+# However native histograms do not represent empty buckets, so here the zeros are implicit.
+eval instant at 5m rate(const_histogram[5m])
+    {} {{schema:0 sum:0 count:0}}
+
+# Zero buckets mean no observations, so average has no meaningful value.
+eval instant at 5m histogram_avg(rate(const_histogram[5m]))
+    {} NaN
+
+# Zero buckets mean no observations, so count is 0.
+eval instant at 5m histogram_count(rate(const_histogram[5m]))
+    {} 0.0
+
+# Zero buckets mean no observations, so the sum should be NaN, However
+# we return 0 for compatibility with classic histograms.
+eval instant at 5m histogram_sum(rate(const_histogram[5m]))
+    {} 0.0
+
+# BUG??? Zero buckets mean no observations, thus any fraction should be 0.
+eval instant at 5m histogram_fraction(0.0, 1.0, rate(const_histogram[5m]))
```

Comment:
That's a good question. We are asking for the fraction of observations among zero observations. So it's a classic 0/0 problem, where NaN is mathematically the correct answer.

As just discussed in person, the problem here is that we ask for a fraction, not the absolute count of observations within the interval. The latter would be no problem, it would just be zero always for an empty histogram. So one way out of this is the introduction of a new function for the absolute count and not the fraction. We could get this as a byproduct of #14651, i.e. `histogram_count(sum(rate(my_histogram[5m])) < 0.1)`.

I would say let's first finish the discussion in #14651. Based on the result, we can make a call if we want a separate function or do the indirect approach as above.

In this PR, I would remove the `BUG???` and maybe describe the problem shortly.